server/svix-server/src/queue/redis.rs

```diff
@@ -181,29 +179,29 @@ async fn background_task(
             ))
             .await?;
 
-        // Acknowledge all the stale ones so the pending queue is cleared
-        let _: RedisResult<()> = pool
-            .query_async(Cmd::xack(&main_queue_name, WORKERS_GROUP, &ids))
-            .await;
+        let mut pipe = redis::pipe();
 
         // And reinsert the map of KV pairs into the MAIN qunue with a new stream ID
         for StreamId { map, .. } in claimed.ids {
-            let _: RedisResult<()> = pool
-                .query_async(Cmd::xadd(
-                    &main_queue_name,
-                    GENERATE_STREAM_ID,
-                    &map.iter()
-                        .filter_map(|(k, v)| {
-                            if let redis::Value::Data(data) = v {
-                                Some((k.as_str(), data.as_slice()))
-                            } else {
-                                None
-                            }
-                        })
-                        .collect::<Vec<(&str, &[u8])>>(),
-                ))
-                .await;
+            let _ = pipe.xadd(
+                &main_queue_name,
+                GENERATE_STREAM_ID,
+                &map.iter()
+                    .filter_map(|(k, v)| {
+                        if let redis::Value::Data(data) = v {
+                            Some((k.as_str(), data.as_slice()))
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<Vec<(&str, &[u8])>>(),
+            );
         }
+
+        // Acknowledge all the stale ones so the pending queue is cleared
+        let _ = pipe.xack(&main_queue_name, WORKERS_GROUP, &ids);
```

Comment:
Can you change this to happen outside of the pipeline and after line 204?

The problem is that a pipeline is not a transaction so it may partially fail, and we want to make sure that xack only pass when the other one passes.